master/internal/webhooks/postgres_webhook.go

```diff
@@ -0,0 +1,55 @@
+package webhooks
+
+import (
+	"context"
+
+	"github.com/uptrace/bun"
+
+	"github.com/determined-ai/determined/master/internal/db"
+)
+
+func AddWebhook(ctx context.Context, w *Webhook) error {
+	return db.Bun().RunInTx(ctx, nil, func(ctx context.Context, tx bun.Tx) error {
+		_, err := tx.NewInsert().Model(w).Exec(ctx)
+		if err != nil {
+			return err
+		}
+		for _, t := range w.Triggers {
+			t.WebhookId = w.ID
+		}
+
+		_, err = tx.NewInsert().Model(&w.Triggers).Exec(ctx)
+		if err != nil {
+			return err
+		}
+		return nil
+	})
+}
+
+func GetWebhooks(ctx context.Context) (Webhooks, error) {
+	webhooks := Webhooks{}
+	err := db.Bun().NewSelect().
+		Model(&webhooks).
+		Relation("Triggers").
+		Scan(ctx)
+	if err != nil {
+		return nil, err
+	}
+	return webhooks, nil
+}
+
+func DeleteWebhook(ctx context.Context, id WebhookID) error {
+	return db.Bun().RunInTx(ctx, nil, func(ctx context.Context, tx bun.Tx) error {
+		_, err := tx.NewDelete().Model((*Webhook)(nil)).Where("id = ?", id).Exec(ctx)
+		if err != nil {
+			return err
+		}
+
+		_, err = tx.NewDelete().Model((*Trigger)(nil)).Where("webhook_id = ?", id).Exec(ctx)
+		if err != nil {
+			return err
+		}
+		return nil
```

Comment:
since you have `ON DELETE CASCADE`, you don't need that 2nd delete query and the wrapping tx. 
by the time webhook is deleted, all it's triggers are already gone.
(if you didn't have `ON DELETE CASCADE`, this trigger deletion would've had to be first - otherwise webhook delete would fail to referential integrity violation)

Reply:
This redundant deletion logic has been removed

Reply:
you can get rid of the `RunInTx` transaction wrapper as well. there's no need for it, since it's a single statement.

Reply:
The transaction wrapper has been removed.